lib/esp-scanning-lib.robot

```diff
@@ -0,0 +1,126 @@
+*** Settings ***
+Documentation       Collection of keywords related to UEFI Secure Boot
+
+Resource            ../keywords.robot
+
+
+*** Variables ***
+${SHIMX64_URL}=     https://cloud.3mdeb.com/index.php/s/TipPHgsCj6MCZHi/download/shimx64.efi
+${GRUBX64_URL}=     https://cloud.3mdeb.com/index.php/s/e2JjYANjMsgyXtS/download/grubx64.efi
+
+
+*** Keywords ***
+Populate /EFI With System Folders
+    [Documentation]    Adds all supported system folders and files to /EFI
+    # given that most of our machines have ubuntu installed,
+    # it's best not to mess with ubuntu EFI files in any way.
+    Add System Folder To /EFI    linux    Fedora
+    Add System Folder To /EFI    manager    Suse
+    Add System Folder To /EFI    windows
+    Add System Folder To /EFI    manager    Redhat
+    Add System Folder To /EFI    linux    Redhat
+    Add System Folder To /EFI    linux    Centos
+    Add System Folder To /EFI    linux    opensuse
+    Add System Folder To /EFI    linux    qubes
+    Add System Folder To /EFI    linux    DTS
+    Add System Folder To /EFI    linux    debian
+
+Add System Folder To /EFI
+    [Documentation]    This keyword creates directories and .efi files for
+    ...    given systems. Available options:
+    ...    ${type} - linux, windows, manager
+    ...    ${double_entry_check} - boolean; specifies whether to download both
+    ...    shim and grub files, ignored for windows type
+    ...    ${folder_name} - usually simply system name, ignored for windows
+    [Arguments]    ${type}    ${folder_name}=Microsoft
+
+    IF    "${type}"=="windows"
+        SSHLibrary.Execute Command    mkdir Microsoft
+        SSHLibrary.Execute Command    mkdir Microsoft/Boot
+        SSHLibrary.Execute Command    wget ${SHIMX64_URL} -O Microsoft/Boot/bootmgfw.efi
+    END
+    IF    "${type}"=="manager"
+        SSHLibrary.Execute Command    mkdir ${folder_name}
+        SSHLibrary.Execute Command    wget ${SHIMX64_URL} -O ${folder_name}/elilo.efi
+    END
+    IF    "${type}"=="linux"
+        SSHLibrary.Execute Command    mkdir ${folder_name}
+        SSHLibrary.Execute Command    wget ${SHIMX64_URL} -O ${folder_name}/shimx64.efi
+    END
+    SSHLibrary.Execute Command    wget ${GRUBX64_URL} -O ${folder_name}/grubx64.efi
+
+    # since switch to root user doesn't really wanna work
+    SSHLibrary.Execute Command    cp -r ~/${folder_name}/ /boot/efi/EFI/
+    ...    sudo=True
+    ...    sudo_password=ubuntu
+
+    SSHLibrary.Execute Command    sync
+    ...    sudo=True
+    ...    sudo_password=ubuntu
+
+    SSHLibrary.Execute Command    rm ${folder_name} -r
+
+Check Boot Menu For All Supported Systems
+    [Documentation]    This keyword scans the boot menu in search for all
+    ...    systems supported by the ESP scan feature. To be used after
+    ...    populating ESP with all the required files.
+    ...    Available modes:
+    ...    normal
+    ...    empty - keyword fails if the list is not empty
```

Comment:
In one case we check the boot menu for all the OSes, in the other we expect the boot menu to be empty - I thought it'd be more efficient to have that in one keyword rather than two mostly overlapping keywords